c10/util/variant.h

```diff
@@ -878,8 +878,8 @@ using std::is_trivially_move_constructible;
 #else
 template <typename T>
 struct is_trivially_copy_constructible
-    : bool_constant<std::is_copy_constructible<T>::value&& __has_trivial_copy(
-          T)> {};
+    : bool_constant<
+          std::is_copy_constructible<T>::value&& __is_trivially_copyable(T)> {};
```

Comment:
Why is this changing?

Reply:
`__has_trivial_copy` is deprecated (I think for a decade) and as of clang version 15 is a build error on my machine. `__is_trivially_copyable` is recommended by the compiler, and supported by both clang and gcc.